content/200-concepts/100-components/02-prisma-client/260-client-extensions/100-client-extensions-model.mdx

```diff
@@ -0,0 +1,135 @@
+---
+title: 'Add custom methods to your models (Prisma Client extensions)'
+metaTitle: 'Prisma Client extensions: model (Preview)'
+metaDescription: 'Extend the functionality of Prisma Client'
+tocDepth: 3
+---
+
+<TopBlock>
+
+<Admonition type="warning">
+
+This page is in draft.
+
+</Admonition>
+
+From version **TBD**, you can use `model` [Prisma Client extensions](/concepts/components/prisma-client/client-extensions) to add custom methods to your models.
+
+To create Prisma Client extensions, use the [`$extends`](/concepts/components/prisma-client/client-extensions#about-prisma-client-extensions) client-level method. To add custom methods to your models, use the `model` extension type.
+
+Possible uses include the following:
+
+- New operations to operate alongside existing Prisma Client operations, such as `findMany`
+- Encapsulated business logic
+- Repetitive operations
+- Model-specific utilities
+
+You can add methods to a specific model in your schema or to all models in your schema.
+
+To create Prisma Client extensions, use the [`$extends`](/concepts/components/prisma-client/client-extensions#about-prisma-client-extensions) client-level method. To add custom methods to your models, use the `model` extension type.
+
+</TopBlock>
+
+## Add the <inlinecode>clientExtensions</inlinecode> feature flag to <inlinecode>schema.prisma</inlinecode>
+
+Before you create Prisma Client extensions, you must enable the `clientExtensions` feature flag in the generator block of your `schema.prisma` file, as follows:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["clientExtensions"]
+}
+```
+
+## Add a custom method to a specific model
+
+To extend a specific model in your schema, use the following structure. This example adds a method to the `user` model.
+
+```ts
+const xprisma = prisma.$extends({
+  name?: '<name>',
+  model?: {
+    user: { ... }   // in this case, we extend the `user` model
+  },
+});
+```
+
+- `name`: an optional field that you can use to name the extension. This name appears in error logs and tracing.
+
+The following example adds a method called `signUp` to the `user` model. This method creates a new user with the specified email address.
+
+```ts
+const xprisma = prisma.$extends({
+  model: {
+    user: {
+      async signUp(email: string) {
+        await prisma.user.create(...)
+      },
+    },
+  }
+})
+```
+
+You would call `signUp` in your application as follows:
+
+```ts
+const user = await xprisma.user.signUp('john@prisma.io')
+```
+
+<Admonition type = "info">
+
+When you call a method in your extension, you must use the correct instance in your code. In the above example, `xprisma.user.signUp` works, but `prisma.user.signUp` does not.
+
+</Admonition>
+
+## Add a custom method to all models in your schema
```

Comment:
```suggestion
## Add a custom method to _all_ models in your schema
```

Reply:
I am not sure I agree that this needs emphasis here, and our general policy is "less is more" with regard to emphasis